contracts/modules/capital/Pool.sol

```diff
@@ -166,44 +196,93 @@ contract Pool is IPool, MasterAware, ReentrancyGuard {
     uint8 decimals,
     uint104 _min,
     uint104 _max,
-    uint16 _maxSlippageRatio
+    uint16 _maxSlippageRatio,
+    bool isPayoutAsset
   ) external onlyGovernance {
-
     require(assetAddress != address(0), "Pool: Asset is zero address");
     require(_max >= _min, "Pool: max < min");
     require(_maxSlippageRatio <= MAX_SLIPPAGE_DENOMINATOR, "Pool: Max slippage ratio > 1");
 
-    uint assetsCount = assets.length;
-    for (uint i = 0; i < assetsCount; i++) {
-      require(assetAddress != assets[i].assetAddress, "Pool: Asset exists");
+    // Check whether the new asset already exists as a payout asset
+    uint payoutAssetsCount = payoutAssets.length;
+    for (uint i = 0; i < payoutAssetsCount; i++) {
+      require(assetAddress != payoutAssets[i].assetAddress, "Pool: Asset exists");
+    }
+
+    // Check whether the new asset already exists as an investment asset
+    uint investmentAssetsCount = investmentAssets.length;
+    for (uint i = 0; i < investmentAssetsCount; i++) {
+      require(assetAddress != investmentAssets[i].assetAddress, "Pool: Asset exists");
     }
```

Comment:
This function prevents us from adding a payout asset if it is already used as an investment asset. For example if we'd (hypothetically) want to add stETH to be used as a payout asset - we'd be unable to do so.

Reply:
You can if you remove it from the investmentAssets.